### PR TITLE
 Implement Order List Prototype and UI Enhancements

### LINKED ISSUE #21: feature/order-list

### PR DESCRIPTION
**Pull request by RaulRinconX**
Description: This Pull Request introduces several features and improvements to the order list functionality and UI, including:

- Order Model: The Order model has been updated to include fields such as title, date, and an optional imageUrl. This allows for more flexibility in displaying order information.

- Order Bloc Implementation: The OrderBloc has been implemented to handle the loading of order data. It manages states like OrdersLoading and OrdersLoaded to control the flow of data between the UI and the business logic.

- OrderItem Widget: The OrderItem widget has been created to display individual orders. It shows the order title, formatted date, and an image if available. It also uses the BasicImage widget for consistent image handling, with a placeholder if no image is provided.

- Historical Order List: The order list is treated as a historical record, so the option to delete orders has been removed to preserve the integrity of past orders.

- UI Enhancements: The overall UI of the order list has been improved with better layout and design consistency, including proper handling of images and text overflow.

## Release Notes

- **New Features**
  - Introduced a new configuration entry for spell checking.
  - Added a `DateUtils` class for formatting order dates.
  - Implemented an `Order` class to manage order details.
  - Enhanced order state management with `OrderBloc`, `OrderItemBloc`, and associated events and states.
  - Created a user interface for displaying orders in `OrderListScreen` and individual order details in `OrderItem`.
  - Added a new `OrderListScreenContent` class for improved UI organization.
  - Enhanced the main application structure to support multiple BLoCs.

- **Bug Fixes**
  - Improved error handling and state management in the order list display.

- **Chores**
  - Added new dependencies for internationalization and nested functionalities.

### CODE REVIEW

File: lib/features/orders/domain/models/order.dart [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    required this.title,
+    required this.date,
+    this.imageUrl,
+  });
+  final String id;
+  final String title;
+  final DateTime date;
+  final String? imageUrl;
+
+  Order copyWith({
```
Comment (abelarismendy) [inline lib/features/orders/domain/models/order.dart:13]: why all params are optional?

File: lib/features/orders/presentation/bloc/order_item_bloc.dart
Comment (abelarismendy) [inline lib/features/orders/presentation/bloc/order_item_bloc.dart:1]: separate this file in three different files. _bloc, _event and _state. Create two folders: './order, .'/order_item

**Review by abelarismendy — CHANGES_REQUESTED** [review]
- Add the new widgets and screen to widgetbook.
- Order item may have more attributes in the future (i.e. state: delivered, in progress, canceled, etc.).
- Separate the bloc, event and state from order_item

File: lib/features/orders/presentation/widgets/order_item.dart [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      shape: RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(24),
+      ),
+      margin: const EdgeInsets.symmetric(vertical: 8, horizontal: 16),
+      child: Padding(
+        padding: const EdgeInsets.all(16),
+        child: Row(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: <Widget>[
+            // Usamos BasicImage para mostrar la imagen
```
Comment (abelarismendy) [inline lib/features/orders/presentation/widgets/order_item.dart]: delete comments

File: lib/features/orders/presentation/widgets/order_item.dart [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            Expanded(
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: <Widget>[
+                  Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                    children: <Widget>[
+                      Expanded(
+                        child: Text(
+                          order.title, // Usamos `title` para el nombre del pedido
```
Comment (abelarismendy) [inline lib/features/orders/presentation/widgets/order_item.dart]: delete comments

File: lib/features/orders/presentation/widgets/order_item.dart [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                  ),
+                  const SizedBox(height: 8),
+                  BlocBuilder<OrderItemBloc, OrderItemState>(
+                    builder: (BuildContext context, OrderItemState state) {
+                      return Row(
+                        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                        crossAxisAlignment: CrossAxisAlignment.end,
+                        children: <Widget>[
+                          Text(
+                            'Delivered: ${DateFormat.yMMMd().format(order.date)}', // Mostramos la fecha formateada
```
[the remark this answers is not in the document]
Comment (abelarismendy) [reply lib/features/orders/presentation/widgets/order_item.dart]: implement this in ./core/utils/
see format_price.dart

[the remark this answers is not in the document]
Comment (abelarismendy) [reply lib/features/orders/presentation/widgets/order_item.dart]: delete comments

**Review by abelarismendy — CHANGES_REQUESTED** [review]

File: lib/features/orders/presentation/widgets/order_item.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (abelarismendy) [reply lib/features/orders/presentation/widgets/order_item.dart]: @RaulRinconX

File: lib/features/orders/domain/models/order.dart (hunk 1 above)
Comment (RaulRinconX) [reply lib/features/orders/domain/models/order.dart:13]: Solved

**Pull request closed without merging**